File: Regular Expressions/Code/chapter4/4-12/recipe4-12.pl
#!/usr/bin/perl -w
use strict;

my $input = $ARGV[0];

if ( $input =~  /^(?:(?:http|ftp)s?):\/\/(?:[A-z0-9][-A-z0-9]+\.)+[A-z]{2,4}$/ )
{
	print "URL '" . $input . "' is valid\n";
} else {
	print "Please enter a valid URL!\n";
}

File: Regular Expressions/Code/chapter4/4-12/recipe4-12.php
<html>
<head><title>4.12 Validating URLs</title></head>
<style>
	.err { color : red ; font-weight : bold }
</style>
<body>
<form action="recipe4-12.php" method="post">
<input type="text" name="input" 
  value="<? print $_POST['input']; ?>" /><br/>
<input type="submit" value="Submit Form" /><br/><br/>
<?php
if ( $_SERVER['REQUEST_METHOD'] == "POST" ) 
{
	$input = $_POST['input'];	
	if ( preg_match( "/^(?:(?:http|ftp)s?):\/\/(?:[A-z0-9][-A-z0-9]+\.)+[A-z]{2,6}$/", $input ) )
	{
		# Do some processing here - input if valid
	}
	else
	{
		print "<span class=\"err\">Bad URL.  Please correct and resubmit the form</span><br/>";
	}
}
?>
</form>
</body>
</html>
